packages/orchestration/src/exos/portfolio-holder-kit.js

```diff
@@ -0,0 +1,200 @@
+import { M, mustMatch } from '@endo/patterns';
+import { E } from '@endo/far';
+import { VowShape } from '@agoric/vow';
+import { Fail } from '@endo/errors';
+import {
+  TopicsRecordShape,
+  PublicTopicShape,
+} from '@agoric/zoe/src/contractSupport/topics.js';
+import { makeScalarBigMapStore } from '@agoric/vat-data';
+
+const { fromEntries } = Object;
+
+/**
+ * @import {MapStore} from '@agoric/store';
+ * @import {VowTools} from '@agoric/vow';
+ * @import {ResolvedPublicTopic, ResolvedTopicsRecord} from '@agoric/zoe/src/contractSupport/topics.js';
+ * @import {Zone} from '@agoric/zone';
+ * @import {CopyMap} from '@endo/patterns';
+ * @import {OrchestrationAccount} from '@agoric/orchestration';
+ * @import {ResolvedContinuingOfferResult} from '../utils/zoe-tools.js';
+ */
+
+/**
+ * @typedef {{
+ *   accounts: MapStore<string, OrchestrationAccount<any>>;
+ *   publicTopics: MapStore<string, ResolvedPublicTopic<unknown>>;
+ * }} PortfolioHolderState
+ */
+
+const ChainNameM = M.string();
+
+const AccountEntriesShape = M.or(
+  M.arrayOf([M.string(), M.remotable('OrchestrationAccount')]),
+  M.mapOf(M.string(), M.remotable('OrchestrationAccount')),
+);
+const PublicTopicsShape = M.or(
+  M.arrayOf([M.string(), PublicTopicShape]),
+  M.mapOf(M.string(), PublicTopicShape),
+);
+
+/**
+ * A kit that holds several OrchestrationAccountKits and returns a invitation
+ * makers.
+ *
+ * @param {Zone} zone
+ * @param {VowTools} vowTools
+ */
+const preparePortfolioHolderKit = (zone, { watch }) => {
+  return zone.exoClassKit(
+    'PortfolioHolderKit',
+    {
+      invitationMakers: M.interface('InvitationMakers', {
+        Action: M.call(ChainNameM, M.string(), M.arrayOf(M.any())).returns(
```

Comment:
I think this can be done by composition, but I don't recall the details.
@michaelfig ? @erights ?

If nothing else, add an XXX plz?

Reply:
I am lacking some context.  What exactly are you indicating by "this" in "this can be done by composition"?

Reply:
In the call `perparePortfolioHolder`, it would be nice to be able to add add'l `invitationMakers`. This would seem to entail:
- include add'l fields in the interface guard
- include add'l methods in the `invitationMakers` facet

Reply:
Timebox expired but hope to revisit soon. Incomplete progress here: https://github.com/Agoric/agoric-sdk/commit/7e886b4c9577ef8c248e7378833e02ef10fa3557